controllers/referee/referee.py

```diff
@@ -1,120 +1,133 @@
+
+import math
+import os
+from controller import Supervisor, AnsiCodes
+
+
+class Referee (Supervisor):
+    def init(self):
+        # create an array of size [3][10] filled in with zeros
+        self.digit = [[0] * 10 for i in range(3)]
+        for j in range(3):
+            for i in range(10):
+                self.digit[j][i] = self.getDevice('digit ' + str(j) + str(i))
+        self.current_digit = [0, 0, 0]  # 0:00
+        self.robot = [0] * 2
+        self.robot[0] = self.getFromDef('WRESTLER_RED')
+        self.robot[1] = self.getFromDef('WRESTLER_BLUE')
+        self.min = [[0] * 3 for i in range(2)]
+        self.max = [[0] * 3 for i in range(2)]
+        for i in range(2):
+            self.min[i] = self.robot[i].getPosition()
+            self.max[i] = self.robot[i].getPosition()
+        self.coverage = [0] * 2
+        self.ko_count = [0] * 2
+        # linear motors on the side of the ring to display the coverage visually
+        self.indicator = [0] * 2
+        self.indicator[0] = self.getDevice('red indicator')
+        self.indicator[1] = self.getDevice('blue indicator')
+
+    def display_time(self, minutes, seconds):
+        for j in range(3):
+            self.digit[j][self.current_digit[j]].setPosition(1000)  # far away, not visible
+        self.current_digit[0] = minutes
+        self.current_digit[1] = seconds // 10
+        self.current_digit[2] = seconds % 10
+        for j in range(3):
+            self.digit[j][self.current_digit[j]].setPosition(0)  # visible
+
+    def run(self, CI):
+        # Performance output used by automated CI script
+        game_duration = 3 * 60 * 1000  # a game lasts 3 minutes
+        # retrieves the WorldInfo.basicTimeTime (ms) from the world file
+        time_step = int(self.getBasicTimeStep())
+        time = 0
+        seconds = -1
+        ko = -1
+        participant = os.environ['PARTICIPANT_NAME'] if 'PARTICIPANT_NAME' in os.environ else 'Participant'
+        opponent = os.environ['OPPONENT_NAME'] if 'OPPONENT_NAME' in os.environ else 'Opponent'
+        self.setLabel(0, '█' * 100, 0, 0, 0.1, 0xffffff, 0.3, 'Lucida Console')
+        self.setLabel(1, '█' * 100, 0, 0.048, 0.1, 0xffffff, 0.3, 'Lucida Console')
+        self.setLabel(2, participant, 0.01, 0.003, 0.08, 0xff0000, 0, 'Arial')
+        self.setLabel(3, opponent, 0.01, 0.051, 0.08, 0x0000ff, 0, 'Arial')
+        while True:
+            if time % 200 == 0:
+                s = int(time / 1000) % 60
+                if seconds != s:
+                    seconds = s
+                    minutes = int(time / 60000)
+                    self.display_time(minutes, seconds)
+                box = [0] * 3
+                for i in range(2):
+                    position = self.robot[i].getPosition()
+                    if abs(position[0]) < 1 and abs(position[1]) < 1:  # inside the ring
+                        coverage = 0
+                        for j in range(3):
```

Comment:
I forgot to correct this: currently, the coverage metric also takes the z axis into account. Maybe we should remove it?
```suggestion
                        for j in range(2):
```

Reply:
Yes, I didn't noticed.